### PR TITLE
Generate static page artefacts via API

### PR DESCRIPTION
**Pull request by kbab**
This PR allows generation of static pages using the API. However, it does not address the recent issue on the use of load_and_annotate_study (except in the static-sitegen/generate_image_page.py script).

### CODE REVIEW

**Comment by liviuba** [comment]
Doesn't look like it's going to be a clean merge, should we rebase on the current main in cases like this?

**Comment by ctr26** [comment]
@liviuba definitely rebase on main before PR, ideally every commit if main is moving fast

File: static-sitegen/scripts/build_collection_site.py
[the remark this answers is not in the document]
Comment (kbab) [reply static-sitegen/scripts/build_collection_site.py:1]: At the time this PR was generated, the BIACollection model did not allow attributes. I believe collection attributes were used in passing the template needed. @liviuba will implement this and then we can enable this feature.

**Comment by ctr26** [comment]
You can force push your rebased branch here that'll work

**Pull request closed without merging**